pkg/bootstrap/bootstrap_config.go

```diff
@@ -254,3 +259,93 @@ func WriteBootstrap(config *meshconfig.ProxyConfig, node string, epoch int, pilo
 	err = t.Execute(fout, opts)
 	return fname, err
 }
+
+// CheckinData has data provided by Pilot needed to start and prepare envoy.
+type CheckinData struct {
+	AvailabilityZone string
+}
+
+var (
+	// IstioCertDir is the location of istio certificates. Visible for testing.
+	// Can be set using ISTIO_CERT_DIR environment variable.
+	IstioCertDir = "/etc/certs"
+
+	// PilotSAN is the DNS name included in the pilot certificate. It may not match the
+	// actual address. Pilot, IstioCA and Sidecar have special certificate that include both
+	// SPIFEE and DNS.
+	PilotSAN = "istio-pilot.istio-system.svc"
+)
+
+// Checkin will connect to pilot and retrieve initial configuration data needed to start envoy.
+// In 0.8 this includes only AZ. Refactored from 'retrieveAZ'.
+// This will retry multiple times.
+func Checkin(secure bool, addr, cluster, node string, delay time.Duration, retries int) (*CheckinData, error) {
+	attempts := 0
+	w := &CheckinData{}
+
+	certDir := os.Getenv("ISTIO_CERT_DIR")
+	if certDir == "" {
+		certDir = IstioCertDir
+	}
+
+	for {
+		var client *http.Client
+		var protocol string
+		if secure {
+			chainCertFile := fmt.Sprintf("%v/%v", certDir, model.CertChainFilename)
+			chainKeyFile := fmt.Sprintf("%v/%v", certDir, model.KeyFilename)
+			chainCert, err := tls.LoadX509KeyPair(chainCertFile, chainKeyFile)
+			if err != nil {
+				log.Infof("Unable to load certs to talk to control plane: %v", err)
+			}
+			caCertFile := fmt.Sprintf("%v/%v", certDir, model.RootCertFilename)
+			caCert, err := ioutil.ReadFile(caCertFile)
+			if err != nil {
+				log.Infof("Unable to load ca root cert to talk to control plane: %v", err)
+			}
+			caCertPool := x509.NewCertPool()
+			caCertPool.AppendCertsFromPEM(caCert)
+
+			//serverName, _, _ := net.SplitHostPort(addr)
+
+			tlsConfig := &tls.Config{
+				Certificates: []tls.Certificate{chainCert},
+				RootCAs:      caCertPool,
+				ServerName:   PilotSAN,
+				// Original ServerName was fmt.Sprintf("%v.svc", serverName), but not sure
```

Comment:
That evaluated to `"istio-pilot.istio-system.svc`, which is what you have in your pilotsan variable.

Reply:
Only in some cases - raw VMs use 'istio-pilot', and for multi-cluster/zVPN the DNS of pilot may be something else.  
We may need to refine this for the case pilot is running in a different namespace. 

There is a PR to make cert generation more flexible - so pilot can have multiple names in the cert.